### DISCUSSION TITLE
Cannot get CF position when flying more than 3 CF with Crazyswarm

### DISCUSSION THREAD

**Discussion by Dhipz** [bitcraze/discussions#343] [Q&A]
Hi, so i'm trying to get the CF position (x, y, z, coordinates) while flying and i was able to do it for up to 3 CF flying simultaneously. But every time i tried using more than 3 CF only 3 of the CF showed up on the rviz and the code i'm running always gives this errors.

```
/home/user/crazyswarm/ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyflie.py:590: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  cfg = yaml.load(crazyflies_yaml)
Warning: No joystick found!
Traceback (most recent call last):
  File "code.py", line 16, in <module>
    c_p = cf.position()
  File "/home/user/crazyswarm/ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyflie.py", line 337, in position
    self.tf.waitForTransform("/world", "/cf" + str(self.id), rospy.Time(0), rospy.Duration(10))
  File "/opt/ros/noetic/lib/python3/dist-packages/tf/listener.py", line 76, in waitForTransform
    raise tf2_ros.TransformException(error_msg or "no such transformation: \"{}\" -> \"{}\"".format(source_frame, target_frame))
tf2.TransformException: canTransform: source_frame cf3 does not exist.
```

This is the test code i used
```py
#!/usr/bin/env python

import numpy as np

from pycrazyswarm import *

swarm = Crazyswarm()
timeHelper = swarm.timeHelper
allcfs = swarm.allcfs


for cf in allcfs.crazyflies:
    cf.takeoff(targetHeight=1.0, duration=2.0)
timeHelper.sleep(3.0)
for cf in allcfs.crazyflies:
    c_p = cf.position()
    print("ID = ", cf.id, " | POS = ", c_p)
    curr_pos = [c_p[0]-0.5,c_p[1],c_p[2]]
    cf.goTo(curr_pos,0,2)
timeHelper.sleep(2.5)
allcfs.land(targetHeight=0.06, duration=2.0)
timeHelper.sleep(3.0)
```

My setup used 2 Lighthouse V2, default CF with lighthouse deck, and all the launch settings are default with the logging enabled and `motion_capture_type` set to none

Sorry for asking it here 🙏

**Comment by knmcguire** [comment]
For reference, the crazyswarm discussion is here: https://github.com/USC-ACTLab/crazyswarm/issues/699

Sorry, we won't be able to help you here, also the entire bitcraze company is in Japan right now. I might be able to look at it in a week, but please remind the crazyswarm maintainers on discussions to help out in the mean time.

**Reply by Dhipz** [reply]
Thank you so much for the reply! I'll try to remind them but please take a look at my problem whenever you can. Once again thank you!

**Comment by bhavya1600** [comment]
> Hi, so i'm trying to get the CF position (x, y, z, coordinates) while flying and i was able to do it for up to 3 CF flying simultaneously. But every time i tried using more than 3 CF only 3 of the CF showed up on the rviz and the code i'm running always gives this errors.
> 
> ```
> /home/user/crazyswarm/ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyflie.py:590: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
>   cfg = yaml.load(crazyflies_yaml)
> Warning: No joystick found!
> Traceback (most recent call last):
>   File "code.py", line 16, in <module>
>     c_p = cf.position()
>   File "/home/user/crazyswarm/ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyflie.py", line 337, in position
>     self.tf.waitForTransform("/world", "/cf" + str(self.id), rospy.Time(0), rospy.Duration(10))
>   File "/opt/ros/noetic/lib/python3/dist-packages/tf/listener.py", line 76, in waitForTransform
>     raise tf2_ros.TransformException(error_msg or "no such transformation: \"{}\" -> \"{}\"".format(source_frame, target_frame))
> tf2.TransformException: canTransform: source_frame cf3 does not exist.
> ```
> 
> This is the test code i used
> 
> ```python
> #!/usr/bin/env python
> 
> import numpy as np
> 
> from pycrazyswarm import *
> 
> swarm = Crazyswarm()
> timeHelper = swarm.timeHelper
> allcfs = swarm.allcfs
> 
> 
> for cf in allcfs.crazyflies:
>     cf.takeoff(targetHeight=1.0, duration=2.0)
> timeHelper.sleep(3.0)
> for cf in allcfs.crazyflies:
>     c_p = cf.position()
>     print("ID = ", cf.id, " | POS = ", c_p)
>     curr_pos = [c_p[0]-0.5,c_p[1],c_p[2]]
>     cf.goTo(curr_pos,0,2)
> timeHelper.sleep(2.5)
> allcfs.land(targetHeight=0.06, duration=2.0)
> timeHelper.sleep(3.0)
> ```
> 
> My setup used 2 Lighthouse V2, default CF with lighthouse deck, and all the launch settings are default with the logging enabled and `motion_capture_type` set to none
> 
> Sorry for asking it here 🙏

Hii Pradhipta,
I am currently trying to get position of just one cf drone, but am unable to do so. Could you help me? It's for my final project.

**Reply by Dhipz** [reply]
Hi, so i have read your problem (i assume its #440) and i don't think i can help you. I have zero knowledge about CF bolt and loco since i only used lighthouse and crazyflie 2.1 for my setup and i used crazyswarm to drive my crazyflie. I hope you  find your solution very soon.
